### PR TITLE
NFT

### PR DESCRIPTION
**Pull request by peterpolman**

### CODE REVIEW

File: contracts/TokenFactory/TokenFactoryFacet.sol [hunk 1]
```diff
@@ -23,7 +41,7 @@ contract TokenFactoryFacet is ITokenFactory {
     ) external override {
         LibDiamond.enforceIsContractOwner();
 
-        TokenLimitedSupply t = new TokenLimitedSupply(_name, _symbol, to, amount);
+        LimitedSupplyToken t = new LimitedSupplyToken(_name, _symbol, to, amount);
```
Comment (bram-rongen) [inline contracts/TokenFactory/TokenFactoryFacet.sol:44]: I was also doubting to do this rename but be aware the name is used in quite some places, al least in deployment, write-abi script, and in the API project... So tread carefully!

Comment (bram-rongen) [reply contracts/TokenFactory/TokenFactoryFacet.sol:44]: It also breaks backwards compatibility, will you release it as 3.0.0? :)

File: contracts/util/ERC721/NonFungibleToken.sol [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+contract NonFungibleToken is ERC721, Ownable {
+    using Counters for Counters.Counter;
+    Counters.Counter private _tokenIds;
+
+    constructor(string memory name_, string memory symbol_, address owner_, string memory baseURI_) ERC721(name_, symbol_) {
+        transferOwnership(owner_);
+        _setBaseURI(baseURI_);
+    }
+
+    function mint(address recipient, string memory tokenURI) external onlyOwner() returns (uint256)
```
Comment (bram-rongen) [inline contracts/util/ERC721/NonFungibleToken.sol]: should modifiers have the `()`? I didn't see that anywhere else..

File: contracts/util/ERC721/NonFungibleToken.sol [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        _tokenIds.increment();
+
+        uint256 newItemId = _tokenIds.current();
+        _mint(recipient, newItemId);
+        _setTokenURI(newItemId, tokenURI);
+
+        return newItemId;
+    }
+}
```
Comment (bram-rongen) [inline contracts/util/ERC721/NonFungibleToken.sol]: newline would be nice

File: contracts/util/ERC721/NonFungibleToken.sol (hunk 2 above)
Comment (peterpolman) [reply contracts/util/ERC721/NonFungibleToken.sol]: Ah, apparently they can, but I was actually not aware of this. You can also pass parameters to modifiers, but when not using that feature, it doesn't make a lot of sense to use the () I guess. I'll remove it!

**Pull request merged**